tests/integration/orcid/test_orcid_converter.py

```diff
@@ -387,3 +390,34 @@ def test_format_thesis_with_author_orcid(app, api_client):
     assert valid_against_schema(result)
     assert xml_compare(expected, result)
     assert _OrcidHasher(mock.Mock())._hash_xml_element(expected) == _OrcidHasher(mock.Mock())._hash_xml_element(result)
+
+
+@pytest.mark.usefixtures('isolated_app')
+class TestBibtexCitation(object):
+    def setup(self):
+        factory = TestRecordMetadata.create_from_file(__name__, 'test_orcid_converter_TestBibtexCitation.json')
+        self.inspire_record = factory.inspire_record
+
+    def test_do_add_bibtex_citation(self):
+        converter = OrcidConverter(
+            record=self.inspire_record,
+            url_pattern='http://inspirehep.net/record/{recid}',
+        )
+        xml_root = converter.get_xml(do_add_bibtex_citation=True)
+        found = False
+        for xml_element in xml_root.getchildren():
+            if 'citation' in xml_element.tag:
+                found = True
+                break
+        if not found:
+            assert False, 'citation not found'
+
+    def test_do_not_add_bibtex_citation(self):
+        converter = OrcidConverter(
+            record=self.inspire_record,
+            url_pattern='http://inspirehep.net/record/{recid}',
+        )
+        xml_root = converter.get_xml()
+        for xml_element in xml_root.getchildren():
+            if 'citation' in xml_element.tag:
+                assert False, 'citation should not be present'
```

Comment:
Wouldn't `assert 'citation' not in xml_element.tag` be more descriptive in case of failure? Also see previous comment on how you can strip out the namespace.

Reply:
Very good point